geode-core/src/main/java/org/apache/geode/management/internal/operation/OperationHistoryManager.java

```diff
@@ -95,13 +95,13 @@ private static boolean isExpired(long expirationTime, OperationState<?, ?> opera
     return operationEnd.getTime() <= expirationTime;
   }
 
-  private OperationState<?, ?> validateLocator(OperationState<?, ?> operationState) {
-    if (isLocatorOffline(operationState)) {
-      operationState.setOperationEnd(new Date(), null,
+  private OperationState<?, ?> validateLocatorAndUpdateOperationState(
+      OperationState<?, ?> operationState) {
+    if (operationState.getOperationEnd() == null && isLocatorOffline(operationState)) {
```

Comment:
get rid of the "getOperationEnd() == null" check here. It is already in isLocatorOffline.

Reply:
Good catch @dschneider-pivotal 
Removed the "operationState.getOperationEnd() == null" in isLocatorOffline()
Thanks.